phalcon/Di/Autowire.zep

```diff
@@ -0,0 +1,160 @@
+    {
+        var constructorParamters, parameter, instanceParameters, autowireType, parameterClass,
+            parameterOptional, parameterDefault, parameterName, parameterHasDefault, useShared, bind;
+
+        if unlikely !class_exists(className) {
+            throw new AutowireException(
+                "Class '" . className . "' wasn't found by autowire compomenent"
+            );
+        }
+
+        let constructorParamters = this->resolveConstructorParameters(className);
+        let useShared = this->useShared;
+
+        if empty constructorParamters {
+            return create_instance(className);
+        }
+
+        let instanceParameters = [];
+
+        for parameter in constructorParamters {
+            let parameterClass = parameter["class"],
+                parameterName = parameter["name"],
+                parameterOptional = parameter["optional"],
+                parameterDefault = parameter["default"],
+                parameterHasDefault = parameter["hasDefault"],
+                autowireType = null;
+
+            if parameterClass {
+                /**
+                 * Check if any bind exists for this class
+                 */
+                if isset autowireTypes[parameterClass] {
+                    let autowireType = autowireTypes[parameterClass];
+                }
+
+                if container->hasBind(parameterClass, autowireType) {
+                    let bind = container->getBind(parameterClass, autowireType);
+                    let instanceParameters[] = bind->resolve(container);
+                } else {
+                    if useShared {
+                        let instanceParameters[] = container->getShared(parameterClass);
+                    } else {
+                        let instanceParameters[] = container->get(parameterClass);
+                    }
+                }
+            } elseif isset parameters[parameterName] {
+                let instanceParameters[] = parameters[parameterName];
+            } elseif parameterHasDefault {
+                let instanceParameters[] = parameterDefault;
+            } else {
+                throw new AutowireException("Missing parameter value for '" . parameterName . "'");
+            }
+        }
+
+        return create_instance_params(className, instanceParameters);
+    }
+
+    protected function resolveConstructorParameters(string! className) -> array
+    {
+        var cache, reflection, constructor, reflectionParameters, parameters, key, parameter;
+
+        let cache = this->cache;
+
+        /**
+         * If className is pretty long let's create sha1 of it to avoid storing long keys while caching
+         */
+        if className->length() > this->maxKeyLength {
+            let className = className->sha1();
+        }
+
+        if isset this->localCache[className] {
+            return this->localCache[className];
+        }
+
+        if cache != null && cache->has("PHAW!" . className) {
```

Comment:
In another components there is prefix that can be specified.

Reply:
> In another components there is prefix that can be specified.

Yea i guess this is not necessary. If user wants will set whatever prefix they want in cache adapter.

Reply:
I removed it.